### ISSUE TITLE
xchm Search in the Left Panel contents section

### ISSUE THREAD

**Issue by aydo-prd** [rzvncj/xCHM#32]
Hello,
In Debian based Pardus operating system
In the chm extension ebook application downloaded from https://sbtransfer.saglik.gov.tr/index.php/s/E6q4pmkAHkF59eQ/download/ekitapV2.chm, Turkish language (ğ,ı,ü, etc.) cannot be searched in the left panel Contents section (For example, the word " broken "). . Is it possible to close all when the tabs are opened in the Contents section of the left panel? Can a Turkish version be added?

Thank you for your effort.

**Comment by rzvncj** [comment]
Your CHM file is not downloadable:
```
$ wget https://sbtransfer.saglik.gov.tr/index.php/s/E6q4pmkAHkF59eQ/download/ekitapV2.chm
--2023-10-31 08:43:12--  https://sbtransfer.saglik.gov.tr/index.php/s/E6q4pmkAHkF59eQ/download/ekitapV2.chm
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving sbtransfer.saglik.gov.tr (sbtransfer.saglik.gov.tr)... 212.175.175.126
Connecting to sbtransfer.saglik.gov.tr (sbtransfer.saglik.gov.tr)|212.175.175.126|:443... connected.
HTTP request sent, awaiting response... 504 Gateway Timeout
Retrying.
```

Having the interface in Turkish is certainly possible, you just need to add a Turkish translation here:

https://github.com/rzvncj/xCHM/tree/master/po

**Comment by aydo-prd** [comment]
https://drive.google.com/file/d/1PwcvwKmlvy3BV2cFO5eNv8No1ACwscgK/view?usp=sharing can you download it again?

**Comment by rzvncj** [comment]
> https://drive.google.com/file/d/1PwcvwKmlvy3BV2cFO5eNv8No1ACwscgK/view?usp=sharing can you download it again?

Yes, that link works, thanks.

**Comment by rzvncj** [comment]
> Hello, In Debian based Pardus operating system In the chm extension ebook application downloaded from https://sbtransfer.saglik.gov.tr/index.php/s/E6q4pmkAHkF59eQ/download/ekitapV2.chm, Turkish language (ğ,ı,ü, etc.) cannot be searched in the left panel Contents section (For example, the word " broken "). . Is it possible to close all when the tabs are opened in the Contents section of the left panel?

I don't understand what you mean by "is it possible to close all when the tabs are opened in the Contents section of the left panel". If you mean that you don't want to see the left-side pane at all, just click on the "Contents" button in the main window, and it will toggle that pane.

Searching your CHM file works just fine, if you use the "Search" tab of the left-side pane (there's "Contents", "Index", and "Search"). The index is not searchable because of how you've built your CHM file. An indexed CHM file contains either a binary index file (`/$WWKeywordLinks/BTree`), or index information in a .hhk text file, and neither are available in your CHM file.

In other words, if you want xCHM to be able to search your index, you need to create an index first when you build the CHM file.

**Comment by aydo-prd** [comment]
https://drive.google.com/file/d/1jI955nZNpdFihcsWDL3PnTrfMZKQ2mW2/view?usp=sharing
I recorded a video. I can explain it better if you run the file with chm extension in Windows operating system.
When we come to the K tab, when we type broken on the keyboard, it goes to the broken tab. While the left side contents tab is open, when I right click and click close all tabs, it closes.


Sir, as you can see in the ekitap.chm application, disease files and information are searched among 1000s of files. XCHM application works very efficiently. Thank you for your hard work.

**Comment by rzvncj** [comment]
I see. You basically want to be able to select a parent node in the contents tree and have xCHM jump to the first child node that begins with the word you're typing on the keyboard while the parent node is selected.

There's no goal for feature parity with hh.exe (the Windows CHM viewer) for xCHM, and this is a feature request (not a bug report), so unfortunately there are no plans to add this functionality to xCHM.

However, this being free / open source, you're welcome to fork the source code, implement the feature and submit a pull request with the changes.

Thanks!

**Comment by aydo-prd** [comment]
I shared a video not for comparison purposes, but to show you how to move quickly through the tabs in the left contents panel and to close the tabs in bulk. When these are added, it will be much better in terms of performance and usage than other applications.

**Comment by rzvncj** [comment]
> I shared a video not for comparison purposes, but to show you how to move quickly through the tabs in the left contents panel and to close the tabs in bulk. When these are added, it will be much better in terms of performance and usage than other applications.

xCHM is a free / open source project with no generating income and a single mainainer (me). I don't, unfortunately, have the time to take on every feature that the Windows viewer offers, and prefer to spend what little time I have on bugs and pull requests for the project, at least in the foreseeable future.

So while anyone is welcome to contribute features they like in a pull request (and I encourage you to do so), I have no plans to implement this feature.

In particular, I also think that similar functionality could be obtained by simply adding an index to your CHM file and using the index tab to search for words you're interested in. While the hh.exe feature is handy, my opinion is that searching directly in the "Contents" tree is bypassing the actual problem with the way the document is designed.
